extras/menus/advancedvsh/advanced.c

```diff
@@ -46,385 +46,416 @@ extern char* available_fonts[];
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0ee82ee;
-							blit_set_color(fc,bc);
-							break;
-						// Pink 
-						case 14:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0x00cbc0ff;
-							blit_set_color(fc,bc);
-							break;
-						// Light Pink 
-						case 15:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0cbc0ff;
-							blit_set_color(fc,bc);
-							break;
-						// Purple 
-						case 16:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0x00993366;
-							blit_set_color(fc,bc);
-							break;
-						// Light Purple 
-						case 17:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0993366;
-							blit_set_color(fc,bc);
-							break;
-						// Teal 
-						case 18:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0x00808000;
-							blit_set_color(fc,bc);
-							break;
-						// Light Teal 
-						case 19:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0808000;
-							blit_set_color(fc,bc);
-							break;
-						// Aqua 
-						case 20:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0x00cccc00;
-							blit_set_color(fc,bc);
-							break;
-						// Light Aqua 
-						case 21:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0cccc00;
-							blit_set_color(fc,bc);
-							break;
-						// Grey 
-						case 22:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0x00737373;
-							blit_set_color(fc,bc);
-							break;
-						// Light Grey 
-						case 23:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0737373;
-							blit_set_color(fc,bc);
-							break;
-						// Black 
-						case 24:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0x00000000;
-							blit_set_color(fc,bc);
-							break;
-						// Light Black 
-						case 25:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa0000000;
-							blit_set_color(fc,bc);
-							break;
-						// Light Red
-						case 26: 
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0xa00000ff;
-							blit_set_color(fc,bc);
-							break;
-						// Red
-						case 27:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0x000000ff;
-							blit_set_color(fc,bc);
-							break;
-						// Light White  
-						case 28:
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0xafffffff;
-							blit_set_color(fc,bc);
-							break;
-						default:	
-							fc = (submax_menu==submenu_sel) ? 0xffffff : 0x0000a5ff;
-							blit_set_color(fc,bc);
-					}
-
-		if(msg) {
-			switch(submax_menu) {
-				case SUBMENU_GO_BACK:
-					xPointer = pointer[2];
-				break;
-				case SUBMENU_RANDOM_GAME:
-					xPointer = 178;
-					break;
-				case SUBMENU_DELETE_HIBERNATION:
-					if (psp_model == PSP_GO)
-						xPointer = 168;
-					else{
-						msg = g_messages[MSG_NO_HIBERNATION];
-						xPointer = 153;
-					}
-					break;
-				case SUBMENU_IMPORT_CLASSIC_PLUGINS:
-				case SUBMENU_ACTIVATE_FLASH_WMA:
-					xPointer = 153;
-					break;
-				default:
-					xPointer=pointer[4];
-					break;
-			}
-
+			// Random
+			case 0:
```

Comment:
I will have to do some test on this, should have a `break` instead of fall through, but that was a previous commit that caused this.